src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -3838,7 +3838,7 @@ private static bool CanReuseParameter(CSharp.Syntax.ParameterSyntax parameter)
                 type = null;
                 name = this.EatToken(SyntaxKind.ArgListKeyword);
             }
-
+            var exclamation = this.CurrentToken.Kind == SyntaxKind.ExclamationToken ? this.EatToken(SyntaxKind.ExclamationToken) : null;
```

Comment:
note.  once https://github.com/dotnet/roslyn/pull/35828 goes in, you'll be able to use `TryEatToken(SyntaxKind.Exclamation)`

Reply:
this applies now.